rclpy/rclpy/qos_event.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from enum import IntEnum
+
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
+from rclpy.waitable import NumberOfEntities
+from rclpy.waitable import Waitable
+
+
+class QoSPublisherEventType(IntEnum):
+    """
+    Enum for QoS events that a Publisher can receive.
+
+    This enum matches the one defined in rcl/event.h
+    """
+
+    RCL_PUBLISHER_OFFERED_DEADLINE_MISSED = 0
+    RCL_PUBLISHER_LIVELINESS_LOST = 1
+
+
+class QoSSubscriptionEventType(IntEnum):
+    """
+    Enum for QoS events that a Subscription can receive.
+
+    This enum matches the one defined in rcl/event.h
+    """
+
+    RCL_SUBSCRIPTION_REQUESTED_DEADLINE_MISSED = 0
+    RCL_SUBSCRIPTION_LIVELINESS_CHANGED = 1
+
+
+QoSLivelinessChangedInfo = namedtuple(
```

Comment:
document these types - is namedtuple ok here? we don't have dataclasses - could use a regular class instead but immutable is nice